supervisor/service.go

```diff
@@ -63,6 +63,14 @@ func (s *Service) CreateContainer(ctx context.Context, r *api.CreateContainerReq
 	}
 	defer func() {
 		if err != nil {
+			clientLog, clientLogErr := client.GetLog(ctx, &shim.GetLogRequest{})
+			if clientLogErr != nil {
+				log.G(s.ctx).WithError(clientLogErr).WithField("container", client.id).
+					Warnf("failed to get shim log")
+			} else {
+				log.G(s.ctx).WithField("container", client.id).
```

Comment:
Would require to unmarshal the logs twice though, once shim side and then again here.

They should rarely be more than 2/3 entries I think.